### PR TITLE
added getting packages by publisher name

### PR DESCRIPTION
**Pull request by mcbaloo**
Implemented when the publisher's name is clicked on the about tab, it fetches all the packages for that particular publisher.

### CODE REVIEW

**Comment by ThinkDigitalSoftware** [comment]
Thanks for your contribution! I'll take a look at this as soon as I can

File: .flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
..._web","path":"C:\\\\Users\\\\Softworks\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\shared_preferences_web-0.1.2+7\\\\","dependencies":[]},{"name":"url_launcher_web","path":"C:\\\\Users\\\\Softworks\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\url_launcher_web-0.1.1+6\\\\","dependencies":[]}]},"dependencyGraph":[{"name":"background_fetch","dependencies":[]},{"name":"flutter_local_notifications","dependencies":[]},{"name":"flutter_secure_storage","dependencies":[]},{"name":"package_info","dependencies":[]},{"name":"path_provider","dependencies":["path_provider_macos","path_provider_linux"]},{"name":"path_provider_linux","dependencies":[]},{"name":"path_provider_macos","dependencies":[]},{"name":"share","dependencies":[]},{"name":"shared_preferences","dependencies":["shared_preferences_macos","shared_preferences_web"]},{"name":"shared_preferences_macos","dependencies":[]},{"name":"shared_preferences_web","dependencies":[]},{"name":"url_launcher","dependencies":["url_launcher_web","url_launcher_macos"]},{"name":"url_launcher_macos","dependencies":[]},{"name":"url_launcher_web","dependencies":[]}],"date_created":"2020-06-15 03:05:17.845620","version":"1.17.3"}
```
Comment (ThinkDigitalSoftware) [inline .flutter-plugins-dependencies:1]: This file is generated. It shouldn't be checked into version control. Can you remove it?

File: lib/main.dart [hunk 2]
```diff
@@ -49,7 +53,11 @@ Future main() async {
         BlocProvider<SubscriptionBloc>(
           create: (BuildContext context) =>
               SubscriptionBloc(client: _htmlParsingClient),
-        )
+        ),
+        // BlocProvider<PublisherBloc>(
```
Comment (ThinkDigitalSoftware) [inline lib/main.dart:57]: If this is longer needed, can it be removed?

File: lib/screens/publisher_package/publisher_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  final PubHtmlParsingClient client;
+  final PageCache _pageCache = PageCache();
+
+  PageRepository({@required this.client});
+
+  Future<Page> get(PageQuery query) async {
+    if (_pageCache.containsKey(query)) {
+      return _pageCache[query];
+    } else {
+      Page page = await client.getPageofPublisherPackages(
```
Comment (ThinkDigitalSoftware) [inline lib/screens/publisher_package/publisher_bloc.dart:135]: Hi, @mcbaloo, does this method exist? It's throwing an error for me and I haven't added this method to the pub client

**Comment by mcbaloo** [comment]
@ThinkDigitalSoftware  are the comments on this thread?

**Review by ThinkDigitalSoftware — COMMENTED** [review]
> @ThinkDigitalSoftware are the comments on this thread?

Yes, they are

File: .flutter-plugins-dependencies (hunk 1 above)
Comment (mcbaloo) [reply .flutter-plugins-dependencies:1]: Alright. I will remove the file and revert back. I understand

File: lib/main.dart (hunk 2 above)
Comment (mcbaloo) [reply lib/main.dart:57]: It is no longer needed. I commented it out. I will refactor and get that sorted.

File: lib/screens/publisher_package/publisher_bloc.dart (hunk 3 above)
Comment (mcbaloo) [reply lib/screens/publisher_package/publisher_bloc.dart:135]: Well, I edited the pub client and added getPageofPublisherPackages to the html_parsing_client to it so as to test. Do I need to provide you with the method? I think that is why you're getting the error.

Comment (ThinkDigitalSoftware) [reply lib/screens/publisher_package/publisher_bloc.dart:135]: Yes, it needs to get added to the package. Did you submit a PR or it was just a change to the local cached package?

Comment (mcbaloo) [reply lib/screens/publisher_package/publisher_bloc.dart:135]: Well, I changed the locally cached package. Maybe I should something a pull request then. Will that be okay?

Comment (ThinkDigitalSoftware) [reply lib/screens/publisher_package/publisher_bloc.dart:135]: Yes, it goes to the [pub_client](https://github.com/ThinkDigitalSoftware/pub_client) repo.
Make sure you add the changes in the Changelog.md and bump the version number

**Pull request merged**